docs/advanced_configuration.md

```diff
@@ -0,0 +1,173 @@
+  )
+rescue WebAuthn::Error => e
+  # Handle error
+end
+```
+
+### Credential Authentication
+
+#### Initiation phase
+
+```ruby
+options = relying_party.options_for_get(allow: user.credentials.map { |c| c.webauthn_id })
+
+# Store the newly generated challenge somewhere so you can have it
+# for the verification phase.
+session[:authentication_challenge] = options.challenge
+
+# Send `options` back to the browser, so that they can be used
+# to call `navigator.credentials.get({ "publicKey": options })`
+
+# You can call `options.as_json` to get a ruby hash with a JSON representation if needed.
+
+# If inside a Rails controller, `render json: options` will just work.
+# I.e. it will encode and convert the options to JSON automatically.
+
+# For your frontend code, you might find @github/webauthn-json npm package useful.
+# Especially for handling the necessary decoding of the options, and sending the
+# `PublicKeyCredential` object back to the server.
+```
+
+#### Verification phase
+
+```ruby
+begin
+  # Assuming you're using @github/webauthn-json package to send the `PublicKeyCredential` object back
+  # in params[:publicKeyCredential]:
+  webauthn_credential, stored_credential = relying_party.verify_authentication(
+    params[:publicKeyCredential],
+    session[:authentication_challenge]
+  ) do
+    # the returned object needs to respond to #public_key and #sign_count
+    user.credentials.find_by(webauthn_id: webauthn_credential.id)
+  end
+
+  # Update the stored credential sign count with the value from `webauthn_credential.sign_count`
+  stored_credential.update!(sign_count: webauthn_credential.sign_count)
+
+  # Continue with successful sign in or 2FA verification...
+
+rescue WebAuthn::SignCountVerificationError => e
+  # Cryptographic verification of the authenticator data succeeded, but the signature counter was less then or equal
+  # to the stored value. This can have several reasons and depending on your risk tolerance you can choose to fail or
+  # pass authentication. For more information see https://www.w3.org/TR/webauthn/#sign-counter
+rescue WebAuthn::Error => e
+  # Handle error
+end
+```
+
+## Moving from Global to Instance Based Configuration
+
+Adding a configuration for a new instance does not mean you need to get rid of your Global configuration. They can co-exist in your application and be both available for the different usages you might have. `WebAuthn.configuration.relying_party` will always return the global one while `WebAuthn::RelyingParty.new` will allow you to create a different instance as you see the need. They will not collide and instead operate in isolation without any shared state.
+
+The gem API described in the current [Usage](../README.md#usage) section for the [Global Configuration](../README.md#configuration) approach will still valid but the [Instance Based API](#instance-based-api) also works with the global `relying_party` that is maintain globally at `WebAuthn.configuration.relying_party`.
```

Comment:
Again, I found this a little bit confusing: where should I create a different instance using `WebAuthn::RelyingParty.new`? Maybe we can benefit of another example here explaining how both configurations co-exists in the `config` file, don't you think?